pkg/proxy/userspace/proxier.go

```diff
@@ -294,6 +296,7 @@ func (proxier *Proxier) addServiceOnPort(service proxy.ServicePortName, protocol
 		return nil, err
 	}
 	si := &serviceInfo{
+		isAlive:             1,
```

Comment:
Are you suggesting that another thread might still see this as a non-1 value?  Or am I missing the implication?  setServiceInfo DOES take the mutex, so any memory write before that must have been visible to other processors.